### ISSUE TITLE
Performance

### ISSUE THREAD

**Issue by ZiadJ** [xenosl/CoroutinesAnywhereInUnity#1]
Thanks for sharing the code for this utility. I'm concerned about performance and how it compares to the built in coroutine so it would probably help if some benchmarks were published on it.

**Comment by xenosl** [comment]
Good point, I'm going to work on it if I have some time (not recently).
But on the other hand, I don't think coroutine is a good choice if performance is critical since new object is created for every yield instruction.
What I'm going to do is just make sure there is no stupid code that make it unreasonable slow.
